pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -200,6 +215,27 @@ func (proxier *metaProxier) OnEndpointSlicesSynced() {
 	proxier.ipv6Proxier.OnEndpointSlicesSynced()
 }
 
+// servicesIPFamily that returns IPFamily of services or error if
+// failed to identify the IP family.
+func servicesIPFamily(service *v1.Service) (*v1.IPFamily, error) {
+	if service == nil {
+		return nil, fmt.Errorf("failed to identify ipfamily for service")
+	}
+	if service.Spec.IPFamily != nil {
+		return service.Spec.IPFamily, nil
+	}
+	if len(service.Spec.ClusterIP) == 0 {
```

Comment:
do we have to consider the case with Spec.ClusterIP = None? or are headless services filtered before we call the proxier?

Reply:
By Spec.ClusterIP = None do you mean Spec.ClusterIP being empty ?

In that case the check on line 232 wouldn't be useful. Therefore error is returned (I think you suggested earlier)

Reply:
headless services literally have ClusterIP = "None" 😅 , 

https://github.com/kubernetes/kubernetes/blob/390a54f737943a1dafa30455c15bb3191cf4d58e/pkg/proxy/iptables/proxier_test.go#L1226-L1229

Reply:
@aojea 
I have added the check for v1.ClusterIPNone